test/integration/query/additional.js

```diff
@@ -869,7 +869,118 @@ module.exports = function (knex) {
+          return knex.raw("WAITFOR DELAY '00:00:10'");
+        },
+        oracledb: function () {
+          return knex.raw('begin dbms_lock.sleep(10); end;');
+        },
+      };
+
+      if (!Object.prototype.hasOwnProperty.call(testQueries, driverName)) {
+        throw new Error('Missing test query for driverName: ' + driverName);
+      }
+
+      const query = testQueries[driverName]();
+
+      function addTimeout() {
+        return query.timeout(200, { cancel: true });
+      }
+
+      // Only mysql/postgres query cancelling supported for now
+      if (
+        !_.startsWith(driverName, 'mysql') &&
+        !_.startsWith(driverName, 'pg')
+      ) {
+        expect(addTimeout).to.throw(
+          'Query cancelling not supported for this dialect'
+        );
+        return; // TODO: Use `this.skip()` here?
+      }
+
+      const getProcessesQueries = {
+        pg: function () {
+          return knex.raw('SELECT * from pg_stat_activity');
+        },
+        mysql: function () {
+          return knex.raw('SHOW PROCESSLIST');
+        },
+        mysql2: function () {
+          return knex.raw('SHOW PROCESSLIST');
+        },
+      };
+
+      if (
+        !Object.prototype.hasOwnProperty.call(getProcessesQueries, driverName)
+      ) {
+        throw new Error('Missing test query for driverName: ' + driverName);
+      }
+
+      const getProcessesQuery = getProcessesQueries[driverName]();
+
+      return knex.transaction((trx) => addTimeout().transacting(trx))
+        .then(function () {
+          expect(true).to.equal(false);
+        })
+        .catch(function (error) {
+          expect(_.pick(error, 'timeout', 'name', 'message')).to.deep.equal({
+            timeout: 200,
+            name: 'KnexTimeoutError',
+            message:
+              'Defined query timeout of 200ms exceeded when running query.',
+          });
+
+          // Ensure sleep command is removed.
+          // This query will hang if a connection gets released back to the pool
+          // too early.
+          // 50ms delay since killing query doesn't seem to have immediate effect to the process listing
+          return delay(50)
+            .then(function () {
+              return getProcessesQuery;
+            })
+            .then(function (results) {
+              let processes;
+              let sleepProcess;
+
+              if (_.startsWith(driverName, 'pg')) {
+                processes = results.rows;
+                sleepProcess = _.find(processes, { query: query.toString() });
+              } else {
+                processes = results[0];
+                sleepProcess = _.find(processes, {
+                  Info: 'SELECT SLEEP(10)',
+                });
+              }
+              expect(sleepProcess).to.equal(undefined);
+            });
+        });
+    });
+
+    it.skip('.timeout(ms, {cancel: true}) should throw error if cancellation cannot acquire connection', async function () {
```

Comment:
We go around the connection pool for the cancellation query by purpose. As the pool will likely to be exhausted when queries start timeouting, which would prevent us from cancelling the query in the first place.

Therefore this test is contradictory and will fail, as an exhausted pool should not prevent the cancellation.

Reply:
We can remove the test then

Reply:
Do we ignore the pool limits for all dialects now, or test still makes sense for some of them?

Reply:
@kibertoad

> Do we ignore the pool limits for all dialects now, or test still makes sense for some of them?

Query cancelling is supported ony for mysql and postgres, so this test was [skipped for other dialects](https://github.com/knex/knex/blob/master/test/integration/query/additional.js#L871-L874) anyway.

```js
      // Only mysql/postgres query cancelling supported for now
      if (!isMysql(knex) && !isPostgreSQL(knex)) {
        return this.skip();
      }
```

> We can remove the test then

Maybe I'd rather turn the test arround that the query is indeed cancelled even if pool is exhausted.

Reply:
Great idea!

Reply:
@kibertoad Test refactored in 944e868